sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -1379,8 +1379,40 @@ pi_result piKernelRelease(pi_kernel) { DIE_NO_IMPLEMENTATION; }
 
 pi_result piEventCreate(pi_context, pi_event *) { DIE_NO_IMPLEMENTATION; }
 
-pi_result piEventGetInfo(pi_event, pi_event_info, size_t, void *, size_t *) {
-  DIE_NO_IMPLEMENTATION;
+pi_result piEventGetInfo(pi_event Event, pi_event_info ParamName,
+                         size_t ParamValueSize, void *ParamValue,
+                         size_t *ParamValueSizeRet) {
+  if (ParamName != PI_EVENT_INFO_COMMAND_EXECUTION_STATUS) {
+    DIE_NO_IMPLEMENTATION;
+  }
+
+  auto CheckAndFillStatus = [&](const cm_support::CM_STATUS &State) {
+    pi_int32 Result = PI_EVENT_RUNNING;
```

Comment:
is this an appropriate default?
@dongkyunahn-intel, your help is needed to review this part. I believe the ``Result` initialization should be a switch on State.

Reply:
It think it would be better to set `Result` with `switch/case` statement as Konst suggested. CM_STATUS_* are defined in link below.

https://github.com/intel/cm-cpu-emulation/blob/0c5fc287f34ae38d3184ab70ea5513d9fb1ff338/common/type_status.h#L13

Reply:
Hello @kbobrovs and @dongkyunahn-intel I did only two states intentionally, because other statuses (!= CM_STATUS_FINISHED) seems logical to map to running since it is all "work in progress" and RT does not need more details. I also used L0 plugin as reference which also has differentiation for two states - finished and not finished = running. I think it would be better to align plugins and not introduce extra value handling on level above. What do you think?
Although it may be needed to report CM_STATUS_RESET separately as error. Could you please educate me what does this status stands for?

Reply:
> Hello @kbobrovs and @dongkyunahn-intel I did only two states intentionally, because other statuses (!= CM_STATUS_FINISHED) seems logical to map to running since it is all "work in progress" and RT does not need more details. I also used L0 plugin as reference which also has differentiation for two states - finished and not finished = running. I think it would be better to align plugins and not introduce extra value handling on level above. What do you think?

I was not aware that L0 has such implementation. Meanwhile, I looked into CM_EMU's `GetStatus()` implementation and found out that the function is dummy one returning only `CM_STATUS_FINISHED`. Now it looks like okay to maintain current implementation - with comments about FINISHED/NOT_FINISHED.

> Although it may be needed to report CM_STATUS_RESET separately as error. Could you please educate me what does this status stands for?

'CM_STATUS_RESET' is defined, but never used in CM_EMU. You can disregard it.